compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -414,15 +414,19 @@ private STNode processEndOfLine() {
     }
 
     /**
-     * Process dot or ellipsis token.
+     * Process dot, ellipsis or decimal floating point token.
      * 
-     * @return Dot or ellipsis token
+     * @return Dot, ellipsis or decimal floating point token
      */
-    private STToken parseDotOrEllipsis() {
+    private STToken processDot() {
         if (reader.peek() == LexerTerminals.DOT && reader.peek(1) == LexerTerminals.DOT) {
             reader.advance(2);
             return getSyntaxToken(SyntaxKind.ELLIPSIS_TOKEN);
         }
+        if (this.mode != ParserMode.IMPORT && isDigit(reader.peek())) {
+            reader.retreat();
+            return processDecimalFloatLiteral();
+        }
```

Comment:
I added `retreat()` method to the `charReader.java`. If it is not acceptable I can mange to do above without calling `reader.retreat();`. But that will add few more lines, if blocks and a new variables to the processNumericLiteral() and processDecimalFloatLiteral(); methods

Reply:
I feel the lexer shouldn't reach here for floating-points.

Here we start parsing a token from a dot. But a floating-point numeric can never start from a dot right?

Reply:
according to spec, `DottedDecimalNumber := DecimalNumber . Digit*  | . Digit+ `
so starting from a `.` is possible.

Reply:
Removed retreat() and refactored. :)